text/2022-02-14-link-application-metrics-with-chaos-experiments.md

```diff
@@ -0,0 +1,212 @@
+
+- Name: custom name of the panel
+- Description: description of the panel
+- Labels: panel could be labeled with many labels, it's powerful to filter the
+  panel
+
+#### Annotations
+
+Annotations are markers on line chart, it could be "instant" or "durable". It
+stands for chaos experiments, events and health check(in the future).
+
+The data of annotations would not be persisted into the storage of Chaos
+Dashboard.
+
+### Available Behaviors for Users
+
+Users could perform following behaviors:
+
+#### Use to a Grafana as "backend"
+
+User could create a Grafana backend with ths host of grafana instance and API
+Keys. As Chaos Dashboard would not modify dashboard in Grafana, so only `Viewer`
+role is enough.
+
+#### Import a Grafana Panel
+
+After the Grafana backend is created, user could import any panel from any
+Grafana dashboard.
+
+From the `share` feature of panel, user would get an url like
+`http://localhost:13000/d/7JdIeTn7z/telegraf-flux?orgId=1&viewPanel=65081`,
+Chaos Dashboard could parse the enough information to identify a Panel:
+
+- Dashboard ID
+- PanelID
+
+Then Chaos Dashboard would ask user to choose which datasource should use, and
+filling the variables which required for expression templating.
+
+At latest, user would be asked to fill the metadata of the imported panel, like
+name, description and labels.
+
+#### View Panels with Chaos Experiment
+
+For any existed chaos experiment, there should be another tab page to access the
+linked panel for current chaos experiment. The "list view" of panels should also
+display preview the data of the linked panel. And user could inspect one of
+panel in more detailed view. Annotation should be displayed on both "list view"
+and "inspect view".
+
+#### Link/Unlink Panels with Chaos Experiment
+
+User could link and unlink another panel at the "list view".
+
+#### Edit Panel
+
+All the fields of the panel should be editable:
+
+- Which Grafana backend to use
+- Dashboard ID
+- Panel ID
+- Name
+- Description
+- Labels
+
+#### Duplicate a Panel
+
+User could duplicate a panel from a existed panel. All the fields of the origin
+panel would be copied to the new panel.
+
+> The feature of duplicate panel is helpful to create series of panels with same
+> Grafana Panel with different variables
+
+#### Remove a Panel
+
+User could remove the panel imported from Grafana. When the panel is removed, it
+would also be removed from the linked chaos experiment.
+
+### About Authorization and Security
+
+Importing panel from Grafana requires a Grafana API Key with role `Viewer`.
+
+Chaos Mesh does not contains any user management system, so all teh Grafana
+backend and imported panels are simply shared with all users.
+
+## Drawbacks
+
+<!-- Why should we not do this? -->
+
+We already have a grafana plugin as mark annotations on grafana dashboard. There
+are also feature overlapping between that grafana plugin and this new feature.
+It might take more effort to keep behaviors consistent.
```

Comment:
Why not add the link feature in the Grafana datasource? Like currently how trace, exemplars, metrics and logs correlations are implemented? This seems less effort and can do the same thing.